identity/src/main/java/com/android/identity/GattClient.java

```diff
@@ -572,28 +579,32 @@ void sendMessage(@NonNull byte[] data) {
             return;
         }
 
-        // Also need room for the leading 0x00 or 0x01.
-        //
-        int maxDataSize = getCharacteristicValueSize() - 1;
-
-        int offset = 0;
-        do {
-            boolean moreDataComing = (offset + maxDataSize < data.length);
-            int size = data.length - offset;
-            if (size > maxDataSize) {
-                size = maxDataSize;
-            }
+        if (data.length == 0) {
+            // Data of length 0 is used to signal we should shut down.
+            mWritingQueue.add(data);
+        } else {
+            // Also need room for the leading 0x00 or 0x01.
+            //
+            int maxDataSize = getCharacteristicValueSize() - 1;
+            int offset = 0;
+            do {
+                boolean moreDataComing = (offset + maxDataSize < data.length);
+                int size = data.length - offset;
+                if (size > maxDataSize) {
+                    size = maxDataSize;
+                }
 
-            byte[] chunk = new byte[size + 1];
-            chunk[0] = moreDataComing ? (byte) 0x01 : (byte) 0x00;
-            System.arraycopy(data, offset, chunk, 1, size);
+                byte[] chunk = new byte[size + 1];
+                chunk[0] = moreDataComing ? (byte) 0x01 : (byte) 0x00;
+                System.arraycopy(data, offset, chunk, 1, size);
 
-            mWritingQueue.add(chunk);
+                mWritingQueue.add(chunk);
```

Comment:
How are `sendMessage` and `drainWritingQueue` meant to work together? It seems like `sendMessage` can add multiple chunks to `mWritingQueue` while `drainWritingQueue` can only remove one chunk at a time.